File: apps/android/core/data/src/main/kotlin/app/gauja/core/data/servers/ServerMapper.kt
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
package app.gauja.core.data.servers

import app.gauja.core.api.models.GetStatus200Response
import app.gauja.core.api.models.PublicSettings
import app.gauja.core.compat.ServerVersion
import app.gauja.core.model.Compatibility
import app.gauja.core.model.ServerAddress
import app.gauja.core.model.ServerSnapshot
import app.gauja.core.model.status.MediaServerType

internal fun mapServer(
    address: ServerAddress,
    status: GetStatus200Response,
    settings: PublicSettings,
): ServerSnapshot =
    ServerSnapshot(
        address,
        status.version,
        settings.applicationTitle,
        settings.initialized,
        status.restartRequired,
        settings.localLogin,
        settings.mediaServerLogin,
        MediaServerType.fromWire(
            settings.mediaServerType
                ?.takeIf { it.isFinite() && it == it.toInt().toDouble() }
                ?.toInt()
        ),
        ServerVersion.parse(status.version)?.compatibility() ?: Compatibility.UNKNOWN,
    )

File: apps/ios/Packages/Data/Sources/Data/servers/ServerMapper.swift
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
import Compat
import Model
internal import SeerrAPI

func mapServer(
    _ address: ServerAddress, _ status: Operations.getStatus.Output.Ok.Body.jsonPayload,
    _ settings: Components.Schemas.PublicSettings
) -> ServerSnapshot {
    let mediaServerType = MediaServerType(rawValue: settings.mediaServerType.flatMap { Int(exactly: $0) })
    return ServerSnapshot(
        address: address, version: status.version, title: settings.applicationTitle,
        initialized: settings.initialized, restartRequired: status.restartRequired,
        localLogin: settings.localLogin, mediaServerLogin: settings.mediaServerLogin,
        mediaServerType: mediaServerType,
        compatibility: ServerVersion(status.version)?.compatibility ?? .unknown)
}
